runbot_send_email/models/runbot_build.py

```diff
@@ -106,141 +97,60 @@ def _subject_email(self):
+                                 self.env.user.name.title(),
+                                 mail_server.smtp_user or ''))
+        reply_to = "{alias}@{domain}".format(
+            alias=config_parameters.get_param('mail.catchall.alias'),
+            domain=config_parameters.get_param('mail.catchall.domain'))
+        emails = self.message_partner_ids.mapped("email")
+        if partner and partner not in self.message_partner_ids:
+            self.message_subscribe_users(user_ids=[partner.user_ids.id])
+        if not emails:
+            _logger.warning('Failed to send the email: Receiver not provided')
+            return
+        values = {'email_from': email_from, 'reply_to': reply_to,
+                  'composition_mode': 'mass_mail'}
+        template = self.env.ref('runbot_send_email.runbot_send_notif')
+        # Render the template
+        return self.sudo().message_post_with_template(template.id, **values)
 
     def _github_status(self):
-        super(RunbotBuild, self)._github_status()
+        build = super(RunbotBuild, self)._github_status()
         for record in self:
             record.get_github_links()
-            record.send_email()
-
-    @api.model
-    def add_followers(self, ids, partners):
-        followers = []
-        build = self.sudo().browse(ids)
-        for partner in partners:
-            followers.append((0, 0, {'res_model': build._name,
-                                     'res_id': build.id,
-                                     'partner_id': partner}))
-        build.write({'message_follower_ids': followers})
-
-    @api.model
-    def select_not_subscribe_partners(self, ids):
-        subscribed = []
-        build = self.browse(ids)
-        followers = build.message_follower_ids
-        if followers:
-            subscribed = followers.mapped('partner_id').ids
-        partner_obj = self.env['res.partner']
-        partner = partner_obj.search([('id', 'not in', subscribed)])
-        partner = partner.filtered(lambda x: x.email)
-        return partner.read()
+            if record.state == 'running':
+                record.send_email()
+        return build
+
+    def update_followers(self):
+        """This method remove or add the user from followers of model
+        'runbot.build' that has logged.
+        """
+        if self.env.user.partner_id not in self.message_partner_ids:
+            self.message_subscribe_users(user_ids=[self.env.uid])
+            follower = True
+        else:
+            self.message_unsubscribe_users(user_ids=[self.env.uid])
+            follower = False
+        return follower
+
+    def create(self, vals):
+        """Add the followers of repository to the followers of build.
+        """
+        build_id = super(RunbotBuild, self).create(vals)
+        users = build_id.repo_id.message_partner_ids.mapped('user_ids')
+        build_id.message_subscribe_users(user_ids=users.ids)
+        return build_id
```

Comment:
Where is the code that adds the git.commiter as follower is the there are a user.email?

Reply:
@moylop260, I do it when sending the email in this [lines](https://github.com/vauxoo-dev/runbot-addons/blob/bce30ed1bde6f846add1d6c14c4b5ae4a37bdd1e/runbot_send_email/models/runbot_build.py#L105-L115), but I think that is better move it to the method `create`, right?